Fix the MallocHeapBreakdown build

rreno opened the pull request:
#### 7b102361b2d70f0dff36d00f9289ef9e5777dedf
<pre>
Fix the MallocHeapBreakdown build
<a href="https://rdar.apple.com/130514774">rdar://130514774</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=275862">https://bugs.webkit.org/show_bug.cgi?id=275862</a>

Reviewed by Yusuke Suzuki.

Fixes the MallocHeapBreakdown debugging build.

* Source/JavaScriptCore/bytecode/ExpressionInfo.cpp:
(JSC::ExpressionInfo::Encoder::createExpressionInfo):
(JSC::ExpressionInfo::createUninitialized):
* Source/JavaScriptCore/heap/GigacageAlignedMemoryAllocator.cpp:
(JSC::GigacageAlignedMemoryAllocator::GigacageAlignedMemoryAllocator):
* Source/JavaScriptCore/heap/IsoAlignedMemoryAllocator.cpp:
(JSC::IsoAlignedMemoryAllocator::IsoAlignedMemoryAllocator):
* Source/JavaScriptCore/wasm/WasmCallee.cpp:
* Source/JavaScriptCore/wasm/WasmCallee.h:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
* Source/WebCore/loader/icon/IconLoader.h:
* Source/bmalloc/bmalloc/IsoMallocFallback.cpp:
(bmalloc::IsoMallocFallback::tryMalloc):
* Tools/TestWebKitAPI/Tests/WTF/WeakPtr.cpp:

Canonical link: <a href="https://commits.webkit.org/280347@main">https://commits.webkit.org/280347@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4f9f58bdce69c75b23554757a6d4414580e43b5d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56349 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35675 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8821 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/59956 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6785 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58475 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43297 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6979 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45640 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4735 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58378 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33552 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48619 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26502 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30332 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5949 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/5789 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/49428 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52327 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6221 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61640 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/55587 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/257 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6347 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52904 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/257 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48685 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/52787 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12477 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/231 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/77346 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31502 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12824 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32588 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33671 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32335 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation commented on Source/bmalloc/bmalloc/IsoMallocFallback.cpp:
```diff
@@ -61,7 +61,7 @@ void determineMallocFallbackState()
 MallocResult tryMalloc(
     size_t size,
-    CompactAllocationMode mode
+    [[maybe_unused]] CompactAllocationMode mode
```
We use `UNUSED_PARAM(mode)` in the body.

rreno replied:
There was some discussion on the mailing list recently and my understanding was we want to move toward `[[maybe_unused]]` instead of the `UNUSED_PARAM` macro. Granted, there was no definitive stance taken so it could go either way!

rreno replied:
https://lists.webkit.org/pipermail/webkit-dev/2024-January/032609.html for reference.

Constellation reviewed (approved):
r=me with nit.

rreno committed:
Fix the MallocHeapBreakdown build
rdar://130514774
https://bugs.webkit.org/show_bug.cgi?id=275862

Reviewed by Yusuke Suzuki.

Fixes the MallocHeapBreakdown debugging build.

* Source/JavaScriptCore/bytecode/ExpressionInfo.cpp:
(JSC::ExpressionInfo::Encoder::createExpressionInfo):
(JSC::ExpressionInfo::createUninitialized):
* Source/JavaScriptCore/heap/GigacageAlignedMemoryAllocator.cpp:
(JSC::GigacageAlignedMemoryAllocator::GigacageAlignedMemoryAllocator):
* Source/JavaScriptCore/heap/IsoAlignedMemoryAllocator.cpp:
(JSC::IsoAlignedMemoryAllocator::IsoAlignedMemoryAllocator):
* Source/JavaScriptCore/wasm/WasmCallee.cpp:
* Source/JavaScriptCore/wasm/WasmCallee.h:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
* Source/WebCore/loader/icon/IconLoader.h:
* Source/bmalloc/bmalloc/IsoMallocFallback.cpp:
(bmalloc::IsoMallocFallback::tryMalloc):
* Tools/TestWebKitAPI/Tests/WTF/WeakPtr.cpp:

Canonical link: https://commits.webkit.org/280347@main
